tailcfg/tailcfg.go

```diff
@@ -2487,6 +2488,11 @@ const (
 	// If multiple values of this key exist, they should be merged in sequence
 	// (replace conflicting keys).
 	NodeAttrServiceHost NodeCapability = "service-host"
+
+	// NodeAttrMaxKeyDuration represents the MaxKeyDuration setting on the
+	// tailnet. The value of this key in [NodeCapMap] will be only one entry of
+	// type [Duration]: an int64 representing the time in nanoseconds.
```

Comment:
`[time.Duration]`?

Reply:
If we're punting this parsing JSON parsing work to clients from a CapMap RawMessage JSON value, that means there are two awkward things for non-Go users:

* a large int64. Some JSON parsers assume all numbers are float64, so we'd only have a range of 104 days: https://go.dev/play/p/ctD4l7VnA69

* time.Duration and nanoseconds: other languages don't have a time.Duration. And nanoseconds is a lot of precision for a key timeout, which we truncate to 1 second on the wire anyway.

Normally in JSON we expression duration as float64 seconds.